pkg/controller/operators/olm/operator.go

```diff
@@ -1327,7 +1328,6 @@ func (a *Operator) transitionCSVState(in v1alpha1.ClusterServiceVersion) (out *v
 
 	modeSet, err := v1alpha1.NewInstallModeSet(out.Spec.InstallModes)
 	if err != nil {
-		syncError = err
 		logger.WithError(err).Warn("csv has invalid installmodes")
 		out.SetPhaseWithEventIfChanged(v1alpha1.CSVPhaseFailed, v1alpha1.CSVReasonInvalidInstallModes, syncError.Error(), now, a.recorder)
```

Comment:
If syncError is being removed here, shouldn't this be changed to err?

Reply:
Yes - great catch.

Reply:
Resolved in the latest version of the PR!